datalake/src/main/java/com/sequenceiq/datalake/service/SdxEventsService.java

```diff
@@ -0,0 +1,145 @@
+import org.springframework.data.domain.Sort;
+import org.springframework.stereotype.Service;
+
+import com.sequenceiq.cloudbreak.api.endpoint.v4.events.EventV4Endpoint;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.events.responses.CloudbreakEventV4Response;
+import com.sequenceiq.cloudbreak.auth.crn.Crn;
+import com.sequenceiq.cloudbreak.auth.crn.CrnParseException;
+import com.sequenceiq.cloudbreak.common.exception.BadRequestException;
+import com.sequenceiq.cloudbreak.structuredevent.event.CloudbreakEventService;
+import com.sequenceiq.cloudbreak.structuredevent.event.StructuredEventType;
+import com.sequenceiq.cloudbreak.structuredevent.event.cdp.CDPOperationDetails;
+import com.sequenceiq.cloudbreak.structuredevent.event.cdp.CDPStructuredEvent;
+import com.sequenceiq.cloudbreak.structuredevent.event.cdp.CDPStructuredNotificationEvent;
+import com.sequenceiq.cloudbreak.structuredevent.service.db.CDPStructuredEventDBService;
+import com.sequenceiq.datalake.entity.SdxCluster;
+import com.sequenceiq.datalake.repository.SdxClusterRepository;
+
+@Service
+public class SdxEventsService {
+
+    @Inject
+    private SdxClusterRepository sdxClusterRepository;
+
+    @Inject
+    private CDPStructuredEventDBService cdpStructuredEventDBService;
+
+    @Inject
+    private EventV4Endpoint eventV4Endpoint;
+
+    public List<CDPStructuredEvent> getDatalakeAuditEvents(String environmentCrn, List<StructuredEventType> types, Integer page, Integer size) {
+        List<CDPStructuredEvent> dlEvents;
+        List<CDPStructuredEvent> cbEvents;
+
+        PageRequest pageable = PageRequest.of(page, size, Sort.by("timestamp").descending());
+        List<String> datalakeCrns = getNonDeletedDatalakeCrns(environmentCrn);
+
+        SdxCluster sdxCluster = getDatalake(environmentCrn);
+        if (sdxCluster == null) {
+            return Collections.EMPTY_LIST;
+        }
+
+        // Retrieve data lake service events
+        Page<CDPStructuredEvent> pagedResponse = cdpStructuredEventDBService.getPagedEventsOfResources(types, datalakeCrns, pageable);
+        if (pagedResponse != null && pagedResponse.getContent() != null && pagedResponse.getContent().size() > 0) {
+            dlEvents = pagedResponse.getContent();
+        } else {
+            dlEvents = Collections.emptyList();
+        }
+
+        // Retrieve cloudbreak service events
+        List<CloudbreakEventV4Response> cloudbreakEventV4Responses = eventV4Endpoint.getCloudbreakEventListByStack(sdxCluster.getName(), page, size,
+                getAccountId(environmentCrn));
+
+        // Translate the cloudbreak events
+        cbEvents = cloudbreakEventV4Responses.stream().map(entry -> convert(entry, sdxCluster.getCrn())).collect(Collectors.toList());
+
+        List<CDPStructuredEvent> combinedEvents = new ArrayList<>();
+        combinedEvents.addAll(cbEvents);
+        combinedEvents.addAll(dlEvents);
+
+        if (combinedEvents.isEmpty()) {
+            return Collections.emptyList();
```

Comment:
Difference between this and `Collections.EMPTY_LIST` used in 53?

Reply:
addressed